script.js

```diff
@@ -0,0 +1,298 @@
+      'CapsLock', 'KeyA', 'KeyS', 'KeyD', 'KeyF', 'KeyG', 'KeyH', 'KeyJ', 'KeyK', 'KeyL', 'Semicolon', 'Quote', 'Enter',
+      'ShiftLeft', 'KeyZ', 'KeyX', 'KeyC', 'KeyV', 'KeyB', 'KeyN', 'KeyM', 'Comma', 'Period', 'Slash', 'ArrowUp', 'ShiftRight',
+      'ControlLeft', 'MetaLeft', 'AltLeft', 'Space', 'AltRight', 'ArrowLeft', 'ArrowDown', 'ArrowRight', 'ControlRight',
+    ];
+  }
+
+  init(wrp, out) {
+    const outputElement = out;
+    this.elements.keysContainer = document.createElement('div');
+    this.elements.keysContainer.classList.add('keyboard');
+    wrp.appendChild(this.elements.keysContainer);
+    const curLang = window.sessionStorage.getItem('lang');
+    this.properties.lang = (curLang == null) ? 'en' : curLang;
+    this.eventHandlers.oninput = (curValue) => {
+      outputElement.value = curValue;
+    };
+    this.createKeys();
+    this.setKeys(this.layouts[this.properties.lang].nocaps);
+    this.elements.keysContainer.addEventListener('kbd', (event) => {
+      const z = this.hardKeys.indexOf(event.detail.code);
+      const key = this.elements.keys[z];
+      const caps = this.properties.capsLock ? 'caps' : 'nocaps';
+      const keyName = this.layouts[this.properties.lang][caps][z];
+      if (keyName !== 'Caps Lock') { key.classList.add('keyboard--key-active'); }
+      switch (keyName) {
+        case 'Backspace':
+          this.properties.value = this.properties.value.slice(0, -1);
+          break;
+        case 'Enter':
+          this.properties.value += '\n';
+          break;
+        case 'Caps Lock':
+          key.classList.toggle('keyboard--key-active');
+          this.toggleCapsLock();
+          break;
+        case 'Shift':
+          this.toggleCapsLock();
+          break;
+        case 'Del':
+          this.properties.value = this.properties.value.slice(1);
+          break;
+        case 'Tab':
+          this.properties.value += '\t';
+          break;
+        case 'Ctrl':
+          if (event.detail.alt) { this.switchLang(); }
+          break;
+        case 'Alt':
+          if (event.detail.ctrl) { this.switchLang(); }
+          break;
+        case 'Win':
+          break;
+        default:
+          this.properties.value += keyName;
+          break;
+      }
+      this.triggerEvent('oninput');
+    });
+
+    this.elements.keysContainer.addEventListener('kbdUp', (event) => {
+      const z = this.hardKeys.indexOf(event.detail.code);
+      const key = this.elements.keys[z];
+      const caps = this.properties.capsLock ? 'caps' : 'nocaps';
+      const keyName = this.layouts[this.properties.lang][caps][z];
+      if (keyName !== 'Caps Lock') { setTimeout(() => { key.classList.remove('keyboard--key-active'); }, 100); }
+      if (keyName === 'Shift') { this.toggleCapsLock(); }
+    });
+    this.elements.keysContainer.addEventListener('click', (event) => {
```

Comment:
Looks like callbacks for click and kbd events are very similar. I think it can be refactored to avoid copy-paste.